daq/host.py

```diff
@@ -47,16 +48,31 @@ class MODE:
     LONG = 'long'
     MERR = 'merr'
 
+def _ip_change_verify(host):
+    if len(host.all_ips) <= 1:
+        return None
+    return len(set(map(lambda ip: ip["ip"], host.all_ips))) > 1
+
+_DEFERRABLE_CHECKS = [
+    {
+        "name": "ipchange",
+        "action": lambda host: host.gateway.execute_script('new_ip', host.target_mac),
+        "verify": _ip_change_verify
+    }
+]
 
 def pre_states():
     """Return pre-test states for basic operation"""
     return ['startup', 'sanity', 'ipaddr', 'base', 'monitor']
 
+def deferrable_check_states():
```

Comment:
how is deferrable different than post?

Also, I think ipchange would need to go near the front of the sequence, since part of what we need to check is that everything else still works after the IP address has changed.  E.g., it's possible (and I've seen it), that the IP stack of the device works, so it responds ot ping -- but an application stack like bacnet will fail (since it cached the original IP address).

Reply:
if the intention is to check whether a new ip has an effect on the device, shouldn't we run it once with one ip, wait for a new ip, then run the test again? I think we can define a set of things a device can ask for when running a test and that can include a new ip, multi dhcp etc through some api on the gateway. When the test is over, gateway can return to default configurations. It'll be easier to do that than making the existing DAQ workflow more complicated.